tests/functional/backend/corpora/test_revisions.py

```diff
@@ -0,0 +1,291 @@
+
+        dataset_1_dropbox_url = "https://www.dropbox.com/s/ib9pth7jr5fvaa8/7MB.h5ad?dl=0"
+        dataset_2_dropbox_url = "https://www.dropbox.com/s/ogempho8wixdlvw/local.h5ad?dl=0"
+
+        # Uploads a dataset
+        self.upload_and_wait(collection_uuid, dataset_1_dropbox_url)
+
+        # make collection public
+        with self.subTest("Test make collection public"):
+            res = requests.post(f"{self.api}/dp/v1/collections/{collection_uuid}/publish", headers=headers)
+            res.raise_for_status()
+            self.assertEqual(res.status_code, requests.codes.accepted)
+
+        dataset_id = requests.get(f"{self.api}/dp/v1/collections/{collection_uuid}").json()['datasets'][0]["id"]
+        explorer_url = self.create_explorer_url(dataset_id)
+        dataset_meta_payload_before_revision = requests.get(f"{self.api}/dp/v1/datasets/meta?url={explorer_url}").json()
+        dataset_schema_before_revision = requests.get(f"{self.api}/cellxgene/e/{dataset_id}/api/v0.2/schema").json()
+
+        with self.subTest("Test updating a dataset in a revision does not effect the published dataset"):
+            # Start a revision
+            res = requests.post(f"{self.api}/dp/v1/collections/{collection_uuid}", headers=headers)
+            self.assertEqual(res.status_code, 201)
+            private_dataset_id = res.json()['datasets'][0]['id']
+
+            # Upload a new dataset
+            new_dataset_id = self.upload_and_wait(collection_uuid, dataset_2_dropbox_url, existing_dataset_id=private_dataset_id)
+
+            # Check that the published dataset is still the same
+            dataset_meta_payload_after_revision = requests.get(f"{self.api}/dp/v1/datasets/meta?url={explorer_url}").json()
+            self.assertDictEqual(dataset_meta_payload_before_revision, dataset_meta_payload_after_revision)
+            dataset_schema_after_revision = requests.get(f"{self.api}/cellxgene/e/{dataset_id}/api/v0.2/schema").json()
+            self.assertDictEqual(dataset_schema_before_revision, dataset_schema_after_revision)
+
+        with self.subTest("Publishing a revised dataset replaces the original dataset"):
+            # Publish the revision
+            res = requests.post(f"{self.api}/dp/v1/collections/{collection_uuid}/publish", headers=headers)
+            res.raise_for_status()
+            self.assertEqual(res.status_code, requests.codes.accepted)
+
+            dataset_meta_payload = requests.get(f"{self.api}/dp/v1/datasets/meta?url={explorer_url}").json()
+            self.assertEqual(dataset_meta_payload["s3_uri"], f"s3://hosted-cellxgene-staging/{new_dataset_id}.cxg/") # TODO
+
+            dataset_schema_old_dataset = requests.get(f"{self.api}/cellxgene/e/{dataset_id}/api/v0.2/schema").json()
+            dataset_schema_new_dataset = requests.get(f"{self.api}/cellxgene/e/{new_dataset_id}/api/v0.2/schema").json()
+            self.assertDictEqual(dataset_schema_old_dataset, dataset_schema_new_dataset)
```

Comment:
Shouldnt these not match if you've replaced the dataset? also once the dataset is replaced `dataset_schema_old_dataset = requests.get(f"{self.api}/cellxgene/e/{dataset_id}/api/v0.2/schema"` shouldnt work (it may continue to work for a few minutes due to caching but I  dont know if we want to depend on that

Reply:
Yes, this part was broken. I deleted it.